include/nlohmann/detail/output/binary_writer.hpp

```diff
@@ -28,6 +28,13 @@ NLOHMANN_JSON_NAMESPACE_BEGIN
 namespace detail
 {
 
+/// how to encode BJData
+enum class bjdata_version_t
+{
+    draft2,
```

Comment:
Didn't you want to use

```cpp
draft2 JSON_HEDLEY_DEPRECATED_FOR(3.12.0, draft3),
```

here?

Reply:
Ah yes, forgot to comment on this - I was getting deprecation warnings just from the default parameter value 😐 Seems the only way to achieve what we want is to overload the function and deprecate the original signature where version isn't specified too. I'll add that later today.